chore(23528): add TODO for linting `@typescript-eslint/await-thenable`

DDDDDanica opened the pull request:
<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been completely filled out, and PR status checks have passed at least once.
-->

## **Description**
This PR addresses TypeScript ESLint error `@typescript-eslint/await-thenabl` across the repository following the planned upgrade to `@metamask/eslint-config` v12.0.0(release notes [here](https://github.com/MetaMask/eslint-config/releases?page=1)). For `@typescript-eslint/await-thenabl` violations:

- User the latest @metamask/eslint-config rule https://github.com/MetaMask/eslint-config/blob/main/packages/typescript/src/index.mjs
- Added eslint-disable directives where proper typing would require significant refactoring
- Each disable directive includes a TODO comment pointing to the tracking issue #31881

This approach follows our strategy for handling ESLint upgrades. The comprehensive fixes for these typing issues will be addressed in dedicated PRs to avoid blocking development.
<!--
Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/32052?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/23528

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

DDDDDanica committed:
chore(23528): add TODO for linting `typescript-eslint/await-thenable`

DDDDDanica committed:
Merge branch 'main' into chore/23528-@typescript-eslint/await-thenable

DDDDDanica committed:
Merge remote-tracking branch 'origin/main' into chore/23528-@typescript-eslint/await-thenable

DDDDDanica committed:
chore: and lint rules to eslint.js and add lint exclude for new files

DDDDDanica committed:
Merge remote-tracking branch 'origin/main' into chore/23528-@typescript-eslint/await-thenable

DDDDDanica committed:
chore: add lint exclude for new files

DDDDDanica committed:
Merge remote-tracking branch 'origin/main' into chore/23528-@typescript-eslint/await-thenable

DDDDDanica committed:
Merge branch 'main' into chore/23528-@typescript-eslint/await-thenable

DDDDDanica committed:
Merge branch 'main' into chore/23528-@typescript-eslint/await-thenable

DDDDDanica committed:
Merge remote-tracking branch 'origin/main' into chore/23528-@typescript-eslint/await-thenable

DDDDDanica committed:
Merge remote-tracking branch 'origin/main' into chore/23528-@typescript-eslint/await-thenable

DDDDDanica committed:
chore: add lint to new files

MajorLift commented on ui/store/actions.ts:
```diff
@@ -3945,6 +3949,8 @@ export function resetOnboarding(): ThunkAction<
   // eslint-disable-next-line @typescript-eslint/no-misused-promises
   return async (dispatch) => {
     try {
+      // TODO: Fix in https://github.com/MetaMask/metamask-extension/issues/31881
+      // eslint-disable-next-line @typescript-eslint/await-thenable
       await dispatch(setSeedPhraseBackedUp(false));
```
I think it might be extract the thunk dispatch calls to a separate issue, since these lint errors are the result of typescript and eslint not being provided the correct types. The correct solution here appears to be updating the `MetaMaskReduxDispatch` type by applying the `useAppDispatch` hook described in the [redux docs](https://redux.js.org/usage/usage-with-typescript#define-typed-hooks), rather than removing the `await` keyword, as is the case with most of the other violations. At the very least, we could update the TODO comments.

DDDDDanica replied:
Thanks for suggestions. It is indeed the correct solution we should aim for, updated in c25d9af834ecd68559989c4d96f51633e9556c43

DDDDDanica replied:
Also applied more to b6d20d0ff741769056b67bc7dec98abae329cbd5

DDDDDanica committed:
chore: add more information to todos for Playwright fixes

DDDDDanica commented on test/e2e/playwright/shared/pageObjects/activity-list-page.ts:
```diff
@@ -18,9 +18,13 @@ export class ActivityListPage {
     const itemStatus = await this.status.innerText();
+    // TODO: Fix in https://github.com/MetaMask/metamask-extension/issues/31881
+    // eslint-disable-next-line @typescript-eslint/await-thenable
     await expect(itemStatus).toEqual('Confirmed');
```
Thanks for thoroughly reviewing, I added some instructions in e54afd5b7ca3e3f5b4c7a7b169d95c23a9d41bcf, it should cover some Playwright Locator & `expect`

DDDDDanica committed:
chore: add more information to todos for withController incorrect promise return

DDDDDanica commented on app/scripts/controllers/account-tracker-controller.test.ts:
```diff
@@ -593,8 +595,14 @@ describe('AccountTrackerController', () => {
         const updateAccountsSpy = jest
           .spyOn(controller, 'updateAccounts')
           .mockResolvedValue();
+        // TODO: Fix in https://github.com/MetaMask/metamask-extension/issues/31881
+        // eslint-disable-next-line @typescript-eslint/await-thenable
         await controller.startPollingByNetworkClientId('networkClientId1');
```
I did some research across the codebase, and it seems like 
- MM uses a synchronous messenger pattern for inter-controller communication
```
const { completedOnboarding } = this.messagingSystem.call(
  'OnboardingController:getState',
);
const selectedAddress = this.messagingSystem.call(
  'AccountsController:getSelectedAccount',
).address;
```

- And the test infrastructure (`withController`) was built assuming everything is async
```
async function withController<ReturnValue>(...args): Promise<ReturnValue> {
  // ... setup
  return await fn({ controller, ... });
}
```

- The codebase likely evolved from async patterns to sync patterns, but tests weren't updated

So what we can do is to add more information on this fix as well, addressed in 9941a9c34f2e3bddf11635c902235be3fa6cdb08

DDDDDanica committed:
chore: extand todos to update `MetaMaskReduxDispatch` type

DDDDDanica committed:
chore: add lint to new files

DDDDDanica committed:
chore: address more codebase with TODO for updating MetaMaskReduxDispatch type

DDDDDanica committed:
Merge remote-tracking branch 'origin/main' into chore/23528-@typescript-eslint/await-thenable

DDDDDanica committed:
Merge remote-tracking branch 'origin/main' into chore/23528-@typescript-eslint/await-thenable

DDDDDanica committed:
chore: add lint todo to new files

DDDDDanica committed:
Merge remote-tracking branch 'origin/main' into chore/23528-@typescript-eslint/await-thenable

DDDDDanica committed:
Merge branch 'main' into chore/23528-@typescript-eslint/await-thenable

DDDDDanica committed:
Merge branch 'main' into chore/23528-@typescript-eslint/await-thenable

DDDDDanica committed:
Merge branch 'main' into chore/23528-@typescript-eslint/await-thenable

DDDDDanica committed:
Merge branch 'main' into chore/23528-@typescript-eslint/await-thenable

DDDDDanica committed:
Merge branch 'main' into chore/23528-@typescript-eslint/await-thenable

DDDDDanica committed:
Merge remote-tracking branch 'origin/main' into chore/23528-@typescript-eslint/await-thenable

DDDDDanica committed:
chore: add lint to new files

DDDDDanica committed:
Merge remote-tracking branch 'origin/main' into chore/23528-@typescript-eslint/await-thenable

DDDDDanica committed:
Merge remote-tracking branch 'origin/main' into chore/23528-@typescript-eslint/await-thenable

DDDDDanica committed:
chore: resolve 2 issues based on cursor bot suggestion

DDDDDanica committed:
Merge branch 'main' into chore/23528-@typescript-eslint/await-thenable

DDDDDanica committed:
Merge branch 'main' into chore/23528-@typescript-eslint/await-thenable

DDDDDanica committed:
Merge branch 'main' into chore/23528-@typescript-eslint/await-thenable

runway-github[bot] committed:
Bump version to 12.22.3 (#34409)

Runway bumped your version to 12.22.3

Co-authored-by: runway-github[bot] <73448015+runway-github[bot]@users.noreply.github.com>

Gudahtt committed:
Initialize changelog

DDDDDanica committed:
Revert "Merge pull request #34065 from MetaMask/Version-v12.23.0"

This reverts commit 24683a112e00d0eac61f04ea2d4d6bf24e781e00, reversing
changes made to f21f2f3207def190ef413aeab7f617c788dc7021.

Gudahtt committed:
Merge pull request #34423 from MetaMask/release/revert-12.23.0

release: Revert "Merge pull request #34065 from MetaMask/Version-v12.23.0"

DDDDDanica committed:
Merge remote-tracking branch 'origin/main' into chore/23528-@typescript-eslint/await-thenable

runway-github[bot] committed:
release(runway): cherry-pick fix: infinite loading on spending cap when selected chain differs from dapp (#34452)

- fix: infinite loading on spending cap when selected chain differs from
dapp cp-12.22.3 (#34355)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**
This PR aims to fix a infinity loading on spending cap confirmation when
a dapp initiate a confirmation in X chain and user selected chain is Y.
<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

[![Open in GitHub

Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/34355?quickstart=1)

## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the
CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing
description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and
accurately)
-->

CHANGELOG entry: Fixed infinite loading on spending cap when selected
chain differs from dapp

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/34298

## **Manual testing steps**

1. Log into Metamask Card site (https://card.metamask.io)
2. Ensure extension network is set to Ethereum mainnet
3. Prompt a spend approval for Linea by "managing limit".
4. Sign "sign in" request.

## **Screenshots/Recordings**



[speding-cap.webm](https://github.com/user-attachments/assets/c0963b52-95c0-405f-bf87-3fb74f84b724)


<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding

Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I’ve applied the right labels on the PR (see [labeling

guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.
[2fb3782](https://github.com/MetaMask/metamask-extension/commit/2fb37826c8060c9309ed93612ce34bbf440e322a)

---------

Co-authored-by: Vinicius Stevam <45455812+vinistevam@users.noreply.github.com>
Co-authored-by: Vinicius Stevam <vinicius.stevam@consensys.net>

DDDDDanica committed:
Merge branch 'main' into chore/23528-@typescript-eslint/await-thenable

Gudahtt committed:
Merge branch 'master' into Version-v12.22.3

DDDDDanica committed:
chore: Update `form-data` to address advisory (#34480) (#34512)

Update `form-data` package to address this advisory:
https://github.com/advisories/GHSA-fjxv-7rqg-78g4

[![Open in GitHub

Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/34480?quickstart=1)

CHANGELOG entry: null

- [x] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding

Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

---------

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/34512?quickstart=1)

## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the
CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing
description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and
accurately)
-->

CHANGELOG entry:

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

---------

Co-authored-by: Mark Stacey <mark.stacey@consensys.net>
Co-authored-by: MetaMask Bot <metamaskbot@users.noreply.github.com>

aganglada committed:
cherry-pick: fix: Reduce Solana snap activity while idle (#34464)

This is a cherry-pick of #34473 for v12.22.3. Original description:

## **Description**

Increase time interval of Solana snap cronjob from 30 minutes to 2
hours, to reduce activity while the wallet is idle.

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/34473?quickstart=1)

## **Changelog**

CHANGELOG entry: Minimize frequent writes while the wallet UI is closed

## **Related issues**

Relates to #33879

## **Manual testing steps**

1. Proceed through onboarding
2. Close all UI instances
3. Monitor the background console, putting a logpoint here conditional
on `this.persist` being `true`:
https://github.com/MetaMask/metamask-extension/blob/f2104e923f463bd85a5950ea030fc00726a319f4/app/scripts/lib/ComposableObservableStore.js#L111

## **Screenshots/Recordings**

N/A

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

runway-github[bot] committed:
release(runway): cherry-pick fix:  Stop polling multichain rates while idle (#34522)

- fix: cp-12.22.3 Stop polling multichain rates while idle (#34474)

## **Description**

The `MultichainRatesController` now stops polling when the UI is closed,
and starts up again when the first UI instance is opened (conditionally,
if a non-EVM account is selected).

The method for starting polling is slightly different from other polling
controllers because this controller doesn't use the same polling mixin
as elsewhere. This controller is deprecated and soon to be removed.

[![Open in GitHub

Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/34474?quickstart=1)

## **Changelog**

CHANGELOG entry: Prevent frequent writes while the wallet UI is closed

## **Related issues**

Relates to #33879

## **Manual testing steps**

1. Proceed through onboarding
2. Close all UI instances
3. Monitor the background console, putting a logpoint here conditional
on `this.persist` being `true`:

https://github.com/MetaMask/metamask-extension/blob/f2104e923f463bd85a5950ea030fc00726a319f4/app/scripts/lib/ComposableObservableStore.js#L111

## **Screenshots/Recordings**

N/A

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding

Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I’ve applied the right labels on the PR (see [labeling

guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.
[49d7ef2](https://github.com/MetaMask/metamask-extension/commit/49d7ef2257b1e8924f77a968ead1adcf2562b683)

Co-authored-by: Mark Stacey <mark.stacey@consensys.net>

Gudahtt committed:
test: Fix flaky account sync test (#34526)

## **Description**

This account sync test was flaky because it wasn't properly waiting for
all account data to finish synchronizing. It was waiting for a single
sync event rather than two. It's unclear why two events are expected,
when this changed (if it did change), or why this wasn't flaky before.
But this seems to make the test consistently pass.

Adding a delay after this point also results in a consistently passing
test, so it's like not a functional bug (since it works if we wait for
syncing to complete).

This change is targeted at the release candidate directly because the
test in question no longer exists on `main`, so we can't fix it there
first.

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/34526?quickstart=1)

## **Changelog**

CHANGELOG entry: null

## **Related issues**

N/A

## **Manual testing steps**

N/A

## **Screenshots/Recordings**

N/A

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

DDDDDanica committed:
release: add changelog for 12.22.3 (#34524)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/34524?quickstart=1)

## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the
CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing
description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and
accurately)
-->

CHANGELOG entry:

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

runway-github[bot] committed:
release(runway): cherry-pick fix:  Fix frequent writes from CronjobController (#34462)

- fix: cp-12.22.3 Fix frequent writes from CronjobController (#34413)

## **Description**

Prevent frequent high-volume writes caused by state updates from the
CronjobController. The controller state includes an event with a
timestamp that can be updated very frequently if the user has a snap
installed with a scheduled recurring event with a short interval. It can
even run while the wallet is otherwise idle, causing the entirety of
wallet state to be written to disk repeatedly.

We have temporarily mitigated the problem by moving the
CronjobController state to a new top-level storage key, so that it can
be written to independently of other state. This controller's state is
very small and should not cause any significant amount of write volume.


[![Open in GitHub

Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/34413?quickstart=1)

## **Changelog**

CHANGELOG entry: Prevent frequent writes while the wallet UI is closed

## **Related issues**

Fixes: #33879

## **Manual testing steps**

To test that cronjobs work correctly:
1. Go to
[test-snaps](https://metamask.github.io/snaps/test-snaps/latest/)
2. Install "Cronjobs Snap".
3. Check if dialog is triggered every 10(?)s. We should update this to
send notifications :sweat_smile:
4. Install "Cronjob Duration Snap".
5. Check if notification is sent every 10(?)s.
6. Install "Background Events Snap".
7. Schedule a background event with a certain date or duration.
8. Check if dialog is triggered after the expected duration.

You can restart the client in between steps to ensure persistence works

To test that write volume has been reduced:
* Monitor disk usage while UI is closed
* Add breakpoint in background console in persistence manager to confirm
there are no writes

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding

Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling

guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

---------

Co-authored-by: Maarten Zuidhoorn <maarten@zuidhoorn.com>
[f2104e9](https://github.com/MetaMask/metamask-extension/commit/f2104e923f463bd85a5950ea030fc00726a319f4)

---------

Co-authored-by: Mark Stacey <mark.stacey@consensys.net>
Co-authored-by: Maarten Zuidhoorn <maarten@zuidhoorn.com>
Co-authored-by: dddddanica <zhaodanica@gmail.com>
Co-authored-by: MetaMask Bot <metamaskbot@users.noreply.github.com>
Co-authored-by: Mark Stacey <markjstacey@gmail.com>

runway-github[bot] committed:
release(runway): cherry-pick fix: Prevent redundant persist store updates (#34516)

- fix: Prevent redundant persist store updates (#34506)

## **Description**

The `ComposableObservableStore` class has been updated to prevent no-op
state updates in cases where a non-persisted property is updated. The
store now introspects patches to see if any of them impact any persisted
property.

[![Open in GitHub

Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/34506?quickstart=1)

## **Changelog**

CHANGELOG entry: Prevent frequent writes while the wallet UI is closed

## **Related issues**

Relates to #33879

## **Manual testing steps**

This problem was discovered when testing
https://github.com/MetaMask/metamask-extension/pull/34465

You can test this by merging the two changes, then following the manual
testing instructions on that PR. I have combined the two PRs here in
this draft for convenience:
https://github.com/MetaMask/metamask-extension/pull/34486

## **Screenshots/Recordings**

N/A

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding

Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I’ve applied the right labels on the PR (see [labeling

guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.
[ac7bc5d](https://github.com/MetaMask/metamask-extension/commit/ac7bc5d3af297de56899dbae08b72813c843a38a)

Co-authored-by: Mark Stacey <mark.stacey@consensys.net>
Co-authored-by: Danica Shen <zhaodanica@gmail.com>

runway-github[bot] committed:
release(runway): cherry-pick fix:  Stop persisting `permissionActivityLog` state (#34523)

- fix: cp-12.22.3 Stop persisting `permissionActivityLog` state (#34465)

## **Description**

Stop persisting `PermissionLogController.permissionActivityLog` state.
This state is not particularly important to persist across sessions, and
it was connected to some performance issues.

[![Open in GitHub

Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/34465?quickstart=1)

## **Changelog**

CHANGELOG entry: Prevent frequent writes while the wallet UI is closed

## **Related issues**

Relates to #33879

## **Manual testing steps**

To test the reduction in writes while idle:

1. Proceed through onboarding
2. Close all UI instances
3. Monitor the background console, putting a logpoint here conditional
on `this.persist` being `true`:

https://github.com/MetaMask/metamask-extension/blob/f2104e923f463bd85a5950ea030fc00726a319f4/app/scripts/lib/ComposableObservableStore.js#L111

~Note that these testing steps will not work without the changes in
https://github.com/MetaMask/metamask-extension/pull/34506~ That is now
merged, and this branch has been updated.

To test the migration:
* Install the extension from a directory with a build of v12.22.2 or
earlier.
* Proceed through onboarding, then wait a few seconds to make sure it
has had enough time to persist
* Update that directory with a build from this PR.
* Start the extension, and review the logs. You should see a log showing
that migration 168.1 was run. If you see a warning that it was skipped,
that suggests something went wrong.
* Check your state to ensure `permissionActivityLog` is not present
(e.g. using `window.stateHooks.getPersistedState()`)

## **Screenshots/Recordings**

N/A

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding

Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I’ve applied the right labels on the PR (see [labeling

guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.
[ee705a1](https://github.com/MetaMask/metamask-extension/commit/ee705a1b6e423a94e7b152658799b04054010191)

---------

Co-authored-by: Mark Stacey <mark.stacey@consensys.net>
Co-authored-by: MetaMask Bot <metamaskbot@users.noreply.github.com>
Co-authored-by: Danica Shen <zhaodanica@gmail.com>
Co-authored-by: Mark Stacey <markjstacey@gmail.com>

Gudahtt committed:
release: 12.22.3 (#34415)

Release candidate for v12.22.3

---------

Co-authored-by: runway-github[bot] <73448015+runway-github[bot]@users.noreply.github.com>
Co-authored-by: Vinicius Stevam <45455812+vinistevam@users.noreply.github.com>
Co-authored-by: Vinicius Stevam <vinicius.stevam@consensys.net>
Co-authored-by: Danica Shen <zhaodanica@gmail.com>
Co-authored-by: MetaMask Bot <metamaskbot@users.noreply.github.com>
Co-authored-by: Alejandro Garcia Anglada <aganglada@gmail.com>
Co-authored-by: Maarten Zuidhoorn <maarten@zuidhoorn.com>

Gudahtt committed:
Version v12.22.3

This is an empty commit to trigger our release scripts.

DDDDDanica committed:
Revert "Revert "Merge pull request #34065 from MetaMask/Version-v12.23.0""

This reverts commit 6bea3c149ec579048eb54675f780ecc5b1be1ef0.

DDDDDanica committed:
chore: Patch cronjob controller based on 34413

metamaskbot committed:
release: dedupe, update LavaMoat policies, update attributions

DDDDDanica committed:
chore: fix yarn dedupe

DDDDDanica committed:
Merge remote-tracking branch 'origin/release/12.23.0-revert' into release/12.23.0-revert

DDDDDanica committed:
chore: fix CHANGELOG.md

DDDDDanica committed:
chore: fix CHANGELOG.md

DDDDDanica committed:
chore: Add @metamask/solana-wallet-snap patch

DDDDDanica committed:
chore: fix @metamask/solana-wallet-snap patch

DDDDDanica committed:
chore: fix @metamask/snap-controllers patch

metamaskbot committed:
release: dedupe, update LavaMoat policies, update attributions

aganglada committed:
fix: solana snap patch

metamaskbot committed:
release: dedupe, update LavaMoat policies, update attributions

DDDDDanica committed:
chore: remove solana package patch legacy

georgewrmarshall reviewed (dismissed):
🟢 Approving changes to component-library/ folder on behalf of @MetaMask/design-system-engineers

Gudahtt committed:
Revert controller-utils bump causing e2e tests to fail

metamaskbot committed:
release: dedupe, update LavaMoat policies, update attributions

DDDDDanica committed:
chore: fix lint issues from merge legacy

DDDDDanica committed:
chore: fix snapshot issues from merge legacy

DDDDDanica committed:
chore: fix snapshot issues from merge legacy

Gudahtt committed:
Undo optimize-media changes and downgrade type package to match main

Gudahtt committed:
Revert extraneous code changes and downgrade accidental bump to snaps-sdk

Gudahtt committed:
Fix lockfile and undo a lot of accidental dependency bumps

metamaskbot committed:
release: dedupe, update LavaMoat policies, update attributions

Gudahtt committed:
Update form-data to resolve advisory

metamaskbot committed:
release: dedupe, update LavaMoat policies, update attributions

Gudahtt committed:
Fix missing dependencies in lockfile

metamaskbot committed:
release: dedupe, update LavaMoat policies, update attributions

Gudahtt committed:
Update unit test snapshot

Gudahtt committed:
Update more test snapshots

Gudahtt committed:
Revert another snapshot change to fix unit test failure

Gudahtt committed:
Undo remaining snapshot changes

Gudahtt committed:
Merge pull request #34583 from MetaMask/release/12.23.0-revert

release: revert changes in master to restore 12.23.0

Gudahtt committed:
Merge pull request #34637 from MetaMask/Version-v12.23.0

Version v12.23.0

DDDDDanica committed:
Merge remote-tracking branch 'origin/master' into chore/23528-@typescript-eslint/await-thenable
